src/main/java/io/cdap/plugin/fileappend/FileAppenderSinkConfig.java

```diff
@@ -0,0 +1,215 @@
+    name = builder.name;
+    outputDir = builder.outputDir;
+    fileNamePrefix = builder.fileNamePrefix;
+    sizeThreshold = builder.sizeThreshold;
+    ageThreshold = builder.ageThreshold;
+    fieldSeparator = builder.fieldSeparator;
+    schema = builder.schema;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(FileAppenderSinkConfig copy) {
+    return new Builder()
+      .setName(copy.getName())
+      .setOutputDir(copy.getOutputDir())
+      .setFileNamePrefix(copy.getFileNamePrefix())
+      .setSizeThreshold(copy.getSizeThreshold())
+      .setAgeThreshold(copy.getAgeThreshold())
+      .setFieldSeparator(copy.getFieldSeparator())
+      .setSchema(copy.getSchema());
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    if (sizeThreshold < 0) {
+      failureCollector.addFailure("Size Threshold must be at least 0.", null)
+        .withConfigProperty(SIZE_THRESHOLD);
+    }
+    if (ageThreshold < 0) {
+      failureCollector.addFailure("Age Threshold must be at least 0.", null)
+        .withConfigProperty(AGE_THRESHOLD);
+    }
+  }
+
+  public void validateOutputSchema(FailureCollector failureCollector, @Nullable Schema inputSchema) {
+    Schema parsedSchema = null;
+    try {
+      parsedSchema = getParsedSchema();
+    } catch (IllegalArgumentException e) {
+      failureCollector.addFailure("Invalid output schema: " + e.getMessage(), null)
+        .withConfigProperty(SCHEMA)
+        .withStacktrace(e.getStackTrace());
+    }
+    if (inputSchema != null && parsedSchema != null) {
+      for (Schema.Field outputField : parsedSchema.getFields()) {
+        Schema.Field inField = inputSchema.getField(outputField.getName());
+        if (inField == null) {
+          failureCollector.addFailure("Field is not in the input schema.", null)
+            .withOutputSchemaField(outputField.getName());
+        } else if (!inField.getSchema().equals(outputField.getSchema())) {
+          failureCollector.addFailure("Schema for field does not match the schema of the input field", null)
+            .withOutputSchemaField(outputField.getName());
+        }
+      }
+    }
+  }
+
+  public String getName() {
+    return name;
+  }
+
+  @Nullable
+  public String getOutputDir() {
+    return outputDir;
+  }
+
+  public String getFileNamePrefix() {
+    return fileNamePrefix;
+  }
+
+  @Nullable
+  public Long getSizeThreshold() {
+    return sizeThreshold;
+  }
+
+  @Nullable
+  public Long getAgeThreshold() {
+    return ageThreshold;
+  }
+
+  @Nullable
+  public String getFieldSeparator() {
+    return fieldSeparator;
+  }
+
+  @Nullable
+  public String getSchema() {
+    return schema;
+  }
+
+  @Nullable
+  public Schema getParsedSchema() {
+    try {
+      return schema != null ? Schema.parseJson(schema) : null;
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Could not parse schema: " + e.getMessage(), e);
```

Comment:
to keep the error message consistent change this to `Invalid schema :  + e.getMessage()` and where it is caught just use `e.getMessage()` as error message